backend/inspirehep/records/api/base.py

```diff
@@ -477,3 +537,27 @@ def revert(self, revision_id):
 
         after_record_revert.send(current_app._get_current_object(), record=self)
         return self
+
+    @property
+    def control_number(self, pid=False):
+        return self.get("control_number")
+
+    @property
+    def control_number_pid(self):
+        return PersistentIdentifier.get(self.pid_type, self.control_number)
+
+    @property
+    def deleted_by(self):
```

Comment:
```suggestion
    def redirected_record_ref(self):
```
and reuse `redirected_pid` property?

Reply:
The idea here was that this returns None when there was no redirection so this is why I'm not reusing `redirected_pid` here.

Reply:
Actually now I can just remove `redirected_pid` as it is not used anymore.